Link error of Optimal_transportation_reconstruction_2_demo when `Cimg` is found

On my machine, a build of all demos fails on one demo:
```
[ 36%] Linking CXX executable Optimal_transportation_reconstruction_2_demo
/usr/bin/ld: CMakeFiles/Optimal_transportation_reconstruction_2_demo.dir/window.cpp.o: undefined reference to symbol 'pthread_mutex_trylock@@GLIBC_2.2.5'
```

My machine has CImg.

Cc @CGAL/geometryfactory and the authors (@cjamin @palliez)

--- a/Optimal_transportation_reconstruction_2/demo/Optimal_transportation_reconstruction_2/CMakeLists.txt
+++ b/Optimal_transportation_reconstruction_2/demo/Optimal_transportation_reconstruction_2/CMakeLists.txt
@@ -30,14 +30,6 @@ find_path(CIMG_INCLUDE_DIR
           DOC "Path to the header of the CImg library")
 
 if (CIMG_INCLUDE_DIR)
-  # Is pthread around? If yes, we need to link against it
-  set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
-  find_package (Threads)
-  if (NOT CMAKE_USE_PTHREADS_INIT) # is it realy pthread?
-    # If not, we're probably on a Windows machine, there is no need to link against anything
-    set(CMAKE_THREAD_LIBS_INIT "")
-  endif()
-
   add_definitions(-DCGAL_USE_CIMG)
   include_directories(${CIMG_INCLUDE_DIR})
   MESSAGE(STATUS "CImg library found, the demo can load point set from image files.")
@@ -98,8 +90,13 @@ if(CGAL_FOUND AND CGAL_Qt5_FOUND AND Qt5_FOUND AND OPENGL_FOUND)
   target_link_libraries( Otr2_demo ${OPENGL_gl_LIBRARY} )
   
   # Link with pthread if necessary
-  if (CMAKE_THREAD_LIBS_INIT)
-    target_link_libraries( Otr2_demo ${CMAKE_THREAD_LIBS_INIT} )
+  if (CIMG_INCLUDE_DIR)
+    # Is pthread around? If yes, we need to link against it
+    set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+    find_package (Threads)
+    if (CMAKE_USE_PTHREADS_INIT)
+      target_link_libraries( Otr2_demo ${CMAKE_THREAD_LIBS_INIT} )
+    endif()
   endif()
     
   add_to_cached_list( CGAL_EXECUTABLE_TARGETS Otr2_demo )

--- a/Optimal_transportation_reconstruction_2/demo/Optimal_transportation_reconstruction_2/CMakeLists.txt
+++ b/Optimal_transportation_reconstruction_2/demo/Optimal_transportation_reconstruction_2/CMakeLists.txt
@@ -30,6 +30,14 @@ find_path(CIMG_INCLUDE_DIR
           DOC "Path to the header of the CImg library")
 
 if (CIMG_INCLUDE_DIR)
+  # Is pthread around? If yes, we need to link against it
+  set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+  find_package (Threads)
+  if (NOT CMAKE_USE_PTHREADS_INIT) # is it realy pthread?
+    # If not, we're probably on a Windows machine, there is no need to link against anything
+    set(CMAKE_THREAD_LIBS_INIT "")
+  endif()
+
   add_definitions(-DCGAL_USE_CIMG)
   include_directories(${CIMG_INCLUDE_DIR})
   MESSAGE(STATUS "CImg library found, the demo can load point set from image files.")
@@ -88,6 +96,11 @@ if(CGAL_FOUND AND CGAL_Qt5_FOUND AND Qt5_FOUND AND OPENGL_FOUND)
 
   # Link with OpenGL
   target_link_libraries( Otr2_demo ${OPENGL_gl_LIBRARY} )
+  
+  # Link with pthread if necessary
+  if (CMAKE_THREAD_LIBS_INIT)
+    target_link_libraries( Otr2_demo ${CMAKE_THREAD_LIBS_INIT} )
+  endif()
     
   add_to_cached_list( CGAL_EXECUTABLE_TARGETS Otr2_demo )
 

--- a/Optimal_transportation_reconstruction_2/include/CGAL/OTR_2/Reconstruction_triangulation_2.h
+++ b/Optimal_transportation_reconstruction_2/include/CGAL/OTR_2/Reconstruction_triangulation_2.h
@@ -795,10 +795,11 @@ public:
     boost::optional<FT> Dqt;
     typename CGAL::cpp11::result_of<typename Traits_::Intersect_2(Line, Line)>::type
       result = intersection(lab, lts);
-    if (result!=boost::none)
+    if (result)
     {
       const Point* iq = boost::get<Point>(&(*result));
-      Dqt = CGAL::approximate_sqrt(geom_traits().compute_squared_distance_2_object()(*iq, pt));
+      if (iq)
+        Dqt = CGAL::approximate_sqrt(geom_traits().compute_squared_distance_2_object()(*iq, pt));
     }
 
     return std::make_pair( (Dabt < FT(0) ? false : true) ,Dqt );

